### PR TITLE
allow building with upstream PortAudio

### LINKED ISSUE #96: Request to remove or replace the PortAudio fork
Audacity vendored and forked PortAudio, including changes that were rejected upstream in 2008. That also involved changes to PortMixer which is also vendored and forked.
https://github.com/audacity/audacity/issues/840

The MIDI support relies on a private API of PortAudio, which is only accessible to link because of the way the vendored PortAudio is hacked into the build system.
https://github.com/audacity/audacity/issues/871

### PR DESCRIPTION
**Pull request by Be-ing**
Resolves: #96

*(short description of the changes and the motivation to make the changes)*

Test this by configuring CMake with ` -Duse_portaudio=system -Duse_portmixer=off`.

This allows Tenacity to work with PipeWire via the JACK API if PortAudio is up to date.

- [x] I have signed off my commits using `-s` or `Signed-off-by`\* (See: [Contributing § DCO](https://github.com/tenacityteam/tenacity/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code\*
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving\*
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"\*

\* indicates required

### CODE REVIEW

**Review by Semisol — DISMISSED** [review]
image
don't know how i can carefully inspect the diff
also someone might want to test if it breaks anything, so don't merge without testing

File: cmake-proxies/CMakeLists.txt [hunk 1]
```diff
@@ -140,7 +140,7 @@ set_conan_vars_to_parent()
 #       directory          option      symbol      req   chk   version
 addlib( libsndfile         sndfile     SNDFILE     YES   YES   "sndfile >= 1.0.28" )
 addlib( libsoxr            soxr        SOXR        YES   YES   "soxr >= 0.1.1" )
-addlib( portaudio-v19      portaudio   PORTAUDIO   YES   YES   "" )
+addlib( portaudio-v19      portaudio   PORTAUDIO   YES   YES   "portaudio-2.0" )
```
Comment (n0toose) [inline cmake-proxies/CMakeLists.txt]: Bit concerned with how all the dependencies have the format of `dependency >= ${MINIMUM_VERSION}`, but this doesn't seem to apply here. Is this surely going to be alright in the long run?

Comment (Be-ing) [reply cmake-proxies/CMakeLists.txt]: Good catch, fixed and rebased.

**Review by n0toose — APPROVED** [review]

**Review by Semisol — APPROVED** [review]

**Comment by parkerlreed** [comment]
Trying on Arch

portaudio from git master

```
  CC=clang cmake \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_INSTALL_PREFIX=/usr \
    -DwxBUILD_TOOLKIT:STRING=gtk3 \
    -Daudacity_use_wxwidgets=local \
    -Duse_portaudio=system \
    -Duse_portmixer=off \
    audacity_use_ffmpeg:STRING=loaded \
    ..
  cmake --build .
  make .
```
```
[ 99%] Linking CXX executable ../bin/Release/audacity
/usr/bin/ld: CMakeFiles/Audacity.dir/AudioIO.cpp.o: in function `MidiTime(void*)':
AudioIO.cpp:(.text+0x37d): undefined reference to `PaUtil_GetTime'
/usr/bin/ld: CMakeFiles/Audacity.dir/AudioIO.cpp.o: in function `AudioIoCallback::AllNotesOff(bool) [clone .constprop.0]':
AudioIO.cpp:(.text+0x4ba): undefined reference to `PaUtil_GetTime'
/usr/bin/ld: CMakeFiles/Audacity.dir/AudioIO.cpp.o: in function `AudioIoCallback::MidiTime()':
AudioIO.cpp:(.text+0x1c27): undefined reference to `PaUtil_GetTime'
/usr/bin/ld: CMakeFiles/Audacity.dir/AudioIO.cpp.o: in function `AudioIoCallback::AllNotesOff(bool)':
AudioIO.cpp:(.text+0x1d17): undefined reference to `PaUtil_GetTime'
/usr/bin/ld: CMakeFiles/Audacity.dir/AudioIO.cpp.o: in function `AudioIoCallback::ComputeMidiTimings(PaStreamCallbackTimeInfo const*, unsigned long)':
AudioIO.cpp:(.text+0x1f5d): undefined reference to `PaUtil_GetTime'
/usr/bin/ld: CMakeFiles/Audacity.dir/AudioIO.cpp.o:AudioIO.cpp:(.text+0x210b): more undefined references to `PaUtil_GetTime' follow
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/Audacity.dir/build.make:7046: bin/Release/audacity] Error 1
make[1]: *** [CMakeFiles/Makefile2:1331: src/CMakeFiles/Audacity.dir/all] Error 2
```

**Comment by Be-ing** [comment]
Try adding `-Duse_portmidi=off`

**Comment by parkerlreed** [comment]
Seems to still be acting weird. Started with a fresh source to make sure I wasn't bringing in anything previous.

```
  mkdir tenacity/build
  cd tenacity/build
  CC=clang cmake \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_INSTALL_PREFIX=/usr \
    -DwxBUILD_TOOLKIT:STRING=gtk3 \
    -Daudacity_use_wxwidgets=local \
    -Duse_portaudio=system \
    -Duse_portmixer=off \
    -Duse_portmidi=off \
    audacity_use_ffmpeg:STRING=loaded \
    ..
  cmake --build .
  make .
```

```
[ 99%] Linking CXX executable ../bin/Release/audacity
/usr/bin/ld: CMakeFiles/Audacity.dir/AudioIO.cpp.o: in function `MidiTime(void*)':
AudioIO.cpp:(.text+0x37d): undefined reference to `PaUtil_GetTime'
/usr/bin/ld: CMakeFiles/Audacity.dir/AudioIO.cpp.o: in function `AudioIoCallback::AllNotesOff(bool) [clone .constprop.0]':
AudioIO.cpp:(.text+0x4ba): undefined reference to `PaUtil_GetTime'
/usr/bin/ld: CMakeFiles/Audacity.dir/AudioIO.cpp.o: in function `AudioIoCallback::MidiTime()':
AudioIO.cpp:(.text+0x1c27): undefined reference to `PaUtil_GetTime'
/usr/bin/ld: CMakeFiles/Audacity.dir/AudioIO.cpp.o: in function `AudioIoCallback::AllNotesOff(bool)':
AudioIO.cpp:(.text+0x1d17): undefined reference to `PaUtil_GetTime'
/usr/bin/ld: CMakeFiles/Audacity.dir/AudioIO.cpp.o: in function `AudioIoCallback::ComputeMidiTimings(PaStreamCallbackTimeInfo const*, unsigned long)':
AudioIO.cpp:(.text+0x1f5d): undefined reference to `PaUtil_GetTime'
/usr/bin/ld: CMakeFiles/Audacity.dir/AudioIO.cpp.o:AudioIO.cpp:(.text+0x210b): more undefined references to `PaUtil_GetTime' follow
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/Audacity.dir/build.make:7046: bin/Release/audacity] Error 1
make[1]: *** [CMakeFiles/Makefile2:1331: src/CMakeFiles/Audacity.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

**Comment by parkerlreed** [comment]
Would the conan cache affect this? Just cleared it and am trying fresh again

EDIT: Nope same result

**Comment by Be-ing** [comment]
The CMake scripts are flaky. You may need to delete the whole build directory and reconfigure to change options. Or you could test #228

**Pull request merged**